helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -281,8 +283,10 @@ private boolean validateOfflineInstancesLimit(final ResourceControllerDataProvid
 
     Map<String, IdealState> newIdealStates = new HashMap<>();
 
+    ClusterConfig clusterConfig = cache.getClusterConfig();
     WagedRebalancer wagedRebalancer =
-        getWagedRebalancer(helixManager, cache.getClusterConfig().getGlobalRebalancePreference());
+        getWagedRebalancer(helixManager, clusterConfig.getGlobalRebalancePreference(),
+            clusterConfig.isGlobalRebalanceAsyncModeEnabled());
```

Comment:
I like the name for this method, `isGlobalRebalanceAsyncModeEnabled` :)

Reply:
That's what you suggested, I also think it is a clean one : )